plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestHiveS3MinioQueries.java

```diff
@@ -105,4 +113,44 @@ private String getDeclaredTableLocation(String tableName)
         }
         throw new IllegalStateException("Location not found in: " + result);
     }
+
+    @Test
+    public void testPathContainsSpecialCharacter()
+    {
+        String tableName = "test_path_special_character" + randomNameSuffix();
+        String location = "s3://%s/%s/".formatted(bucketName, tableName);
+        assertUpdate(format(
+                "CREATE TABLE %s (id bigint, part varchar) WITH (partitioned_by = ARRAY['part'], external_location='%s')",
+                tableName,
+                location));
+
+        assertUpdate("INSERT INTO " + tableName + " VALUES " +
+                "(1, 'with-hyphen')," +
+                "(2, 'with.dot')," +
+                "(3, 'with:colon')," +
+                "(4, 'with/slash')," +
+                "(5, 'with\\\\backslashes')," +
+                "(6, 'with\\backslash')," +
+                "(7, 'with=equal')," +
+                "(8, 'with?question')," +
+                "(9, 'with!exclamation')," +
+                "(10, 'with%percent')," +
+                "(11, 'with%%percents')," +
+                "(12, 'with space')", 12);
+
+        assertQuery("SELECT * FROM " + tableName,
+                "VALUES (1, 'with-hyphen')," +
+                        "(2, 'with.dot')," +
+                        "(3, 'with:colon')," +
+                        "(4, 'with/slash')," +
+                        "(5, 'with\\\\backslashes')," +
+                        "(6, 'with\\backslash')," +
+                        "(7, 'with=equal')," +
+                        "(8, 'with?question')," +
+                        "(9, 'with!exclamation')," +
+                        "(10, 'with%percent')," +
+                        "(11, 'with%%percents')," +
+                        "(12, 'with space')");
```

Comment:
We could extract a variable for the same values in INSERT and SELECT statements.